library/src/main/java/org/apache/apex/malhar/lib/fs/redshift/RedshiftJdbcTransactionableOutputOperator.java

```diff
@@ -0,0 +1,284 @@
+
+import java.sql.BatchUpdateException;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.apex.malhar.lib.fs.FSRecordCompactionOperator;
+
+import com.google.common.base.Preconditions;
+
+import com.datatorrent.api.Context;
+import com.datatorrent.lib.db.jdbc.AbstractJdbcTransactionableOutputOperator;
+
+/**
+ *  A concrete implementation of AbstractJdbcTransactionableOutputOperator for Redshift which takes FSRecordCompactionOperator.OutputMetaData.
+ *  Load the data into the specified redshift tables from data files. The files can be located either in S3 or an Amazon EMR.
+ *  Specify the bucketName property if the file is located in S3 or specify the emrClusterId if the file is location in EMR.
+ *  By default, it load files from S3 into Redshfit table. If the file is located in EMR, then specify "readFromS3" parameter to false.
+ *
+ * @displayName Redshift Output Operator
+ * @category Output
+ * @tags database, jdbc, redshift
+ */
+public class RedshiftJdbcTransactionableOutputOperator extends AbstractJdbcTransactionableOutputOperator<FSRecordCompactionOperator.OutputMetaData>
+{
+  private static final Logger logger = LoggerFactory.getLogger(RedshiftJdbcTransactionableOutputOperator.class);
+  protected static final String DEFAULT_REDSHIFT_DELIMITER = "|";
+  @NotNull
+  private String tableName;
+  @NotNull
+  private String accessKey;
+  @NotNull
+  private String secretKey;
+  @NotNull
+  private String redshiftDelimiter = DEFAULT_REDSHIFT_DELIMITER;
+  private String region;
+  private boolean readFromS3 = true;
+  private String emrClusterId;
+  private String bucketName;
+  protected transient Statement stmt;
+
+  @Override
+  public void setup(Context.OperatorContext context)
+  {
+    super.setup(context);
+    try {
+      stmt = store.getConnection().createStatement();
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  protected String getUpdateCommand()
+  {
+    throw new UnsupportedOperationException("Unsupported Operation");
+  }
+
+  // Preparedstatement is not needed for uploading data into redshift. So, nothing to be done in activate state.
+  @Override
+  public void activate(Context.OperatorContext context)
+  {
+  }
+
+  /**
+   * Create the copy statement from the specified OutputMetaData
+   * @param data Given OutputMetaData
+   * @return the copy statement
+   */
+
+  protected String generateCopyStatement(FSRecordCompactionOperator.OutputMetaData data)
+  {
+    String file = data.getPath();
+    StringBuilder exec = new StringBuilder();
+    exec.append("COPY " + tableName + " ");
+    if (readFromS3) {
+      exec.append("FROM 's3://" + bucketName + "/" + file + "' ");
+    } else {
+      exec.append("FROM 'emr://" + emrClusterId + "/" + file + "' ");
```

Comment:
http://docs.aws.amazon.com/redshift/latest/dg/r_COPY_command_examples.html#copy-command-examples-emr

http://docs.aws.amazon.com/redshift/latest/dg/r_COPY_command_examples.html#sub-example-load-favorite-movies
These examples have "token" or iam_role in the command. Is it mandatory ?